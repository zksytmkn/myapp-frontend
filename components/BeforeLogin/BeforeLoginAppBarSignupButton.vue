<template>
  <v-btn
    outlined
    class="ml-2 font-weight-bold"
    color="appblue"
    to="/signup"
  >
    {{ $my.pageTitle('signup') }}
  </v-btn>
</template>

<script>
export default {
}
</script>