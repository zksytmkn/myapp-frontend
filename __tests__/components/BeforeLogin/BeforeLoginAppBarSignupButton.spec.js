import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import BeforeLoginAppBarSignupButton from '@/components/BeforeLogin/BeforeLoginAppBarSignupButton.vue';

const localVue = createLocalVue();

describe('BeforeLoginAppBarSignupButton.vue', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();

    // $my オブジェクトをモック
    const $my = {
      pageTitle: jest.fn().mockReturnValue('Signup')
    };

    // コンポーネントをマウントする
    wrapper = shallowMount(BeforeLoginAppBarSignupButton, {
      localVue,
      vuetify,
      mocks: {
        $my
      },
      stubs: {
        'before-login-app-bar-signup-button': true,
        'v-btn': true
      }
    });
  });

  it('renders correctly', () => {
    // コンポーネントが正しくレンダリングされているか確認
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has the correct default properties', () => {
    // デフォルトのプロパティが正しいか確認
    const button = wrapper.find('v-btn-stub');
    expect(button.classes()).toContain('ml-2');
    expect(button.classes()).toContain('font-weight-bold');
    expect(button.attributes('color')).toBe('blue');
    expect(button.attributes('outlined')).toBe('true');
  });

  it('has a slot with page title', () => {
    // slotの中にpageTitleが正しく表示されているか確認
    const slotContent = wrapper.find('v-btn-stub').html();
    expect(slotContent).toContain('Signup');
  });

  it('navigates to the correct page on click', () => {
    // クリックしたときに正しいページに遷移するか確認
    const button = wrapper.find('v-btn-stub');
    expect(button.attributes('to')).toBe('/signup');
  });
});
