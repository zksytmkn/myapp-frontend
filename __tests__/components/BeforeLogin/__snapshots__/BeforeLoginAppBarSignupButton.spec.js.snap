// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeforeLoginAppBarSignupButton.vue renders correctly 1`] = `
<v-btn-stub
  activeclass=""
  class="ml-2 font-weight-bold"
  color="blue"
  outlined="true"
  tag="button"
  to="/signup"
  type="button"
>
  
    Signup
  
</v-btn-stub>
`;
